libraries/DataFlash/LogFile.cpp

```diff
@@ -1799,24 +1799,38 @@ void DataFlash_Class::Log_Write_AttitudeView(AP_AHRS_View &ahrs, const Vector3f
 void DataFlash_Class::Log_Write_Current(const AP_BattMonitor &battery)
 {
     if (battery.num_instances() >= 1) {
+        float temp;
+        bool has_temp = battery.get_temperature(temp, 0);
         struct log_Current pkt = {
             LOG_PACKET_HEADER_INIT(LOG_CURRENT_MSG),
             time_us             : AP_HAL::micros64(),
             battery_voltage     : battery.voltage(0),
             current_amps        : battery.current_amps(0),
             current_total       : battery.current_total_mah(0),
+            temperature         : (int16_t)(has_temp ? (temp * 100) : 0),
         };
+        AP_BattMonitor::cells cells = battery.get_cell_voltages(0);
+        for (uint8_t i = 0; i < MAVLINK_MSG_BATTERY_STATUS_FIELD_VOLTAGES_LEN; i++) {
```

Comment:
It looks reasonable to me, but then this happens:

```
../../libraries/DataFlash/LogFile.cpp:1813:77: error: cannot bind packed field ‘pkt.log_Current::cell_voltages’ to ‘short unsigned int (&)[10]’
         for (uint8_t i = 0; i < MIN(ARRAY_SIZE(cells.cells), ARRAY_SIZE(pkt.cell_voltages)); i++) {
                                                                         ~~~~^
../../libraries/AP_Common/AP_Common.h:69:52: note: in definition of macro ‘ARRAY_SIZE’
 #define ARRAY_SIZE(_arr) sizeof(_ARRAY_SIZE_HELPER(_arr))
                                                    ^~~~
```

Lazy version would be to simply make the DF log length always match the AP_BattMonitor::cells.cells length?